src/java.desktop/unix/classes/sun/awt/X11/XWM.java

```diff
@@ -1369,6 +1369,9 @@ Insets guessInsets(XDecoratedPeer window) {
               case UNITY_COMPIZ_WM:
                   res = new Insets(28, 1, 1, 1);
                   break;
+              case MUTTER_WM:
+                  res = new Insets(37, 0, 0, 0);
```

Comment:
How did you come up with the numbers for these insets?

Reply:
They are obtained from the system, and are a default value for the Gnome shell (which uses Mutter WM).
You can get them with `window.getInsets()`.